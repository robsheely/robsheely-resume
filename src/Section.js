import React, { Fragment } from "react";
import PropTypes from "prop-types";

import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Divider from "@material-ui/core/Divider";
import Typography from "@material-ui/core/Typography";
import { withStyles } from "@material-ui/core";

const styles = theme => ({
  card: {
    marginTop: 10,
    marginBottom: 10,
    backgroundColor: theme.palette.secondary[ "50" ]
  },
  content: {
    textAlign: "center"
  },
  label: {
    color: theme.palette.secondary[ "600" ]
  },
  divider: {
    marginTop: 6,
    marginBottom: 16,
    backgroundColor: theme.palette.primary.main
  },
  h6: {
    fontWeight: 500
  }
});

const Section = props => {
  const { classes, label, children } = props;
  // Convert spaces to dashes and all chars to lowercase to work as part of ids
  const name = (label) ? label.replace(/\s+/g, "-").toLowerCase() : "none";

  return (
    <div id={`section-${name}`} data-testid={`section-${name}`}>
      <Card className={classes.card}>
        <CardContent
          id={`section-${name}-content`}
          data-testid={`section-${name}-content`}
          classes={{ root: classes.content }}
        >
          {
            label &&
            <Fragment>
              <Typography
                id={`section-${name}-content-label`}
                data-testid={`section-${name}-content-label`}
                variant="h6"
                color="inherit"
                classes={{
                  root: classes.label,
                  h6: classes.h6
                }}>
                {label}
              </Typography>
              <Divider
                id={`section-${name}-content-divider`}
                data-testid={`section-${name}-content-divider`}
                classes={{ root: classes.divider }}
              />
            </Fragment>
          }
          {children}
        </CardContent>
      </Card>
    </div>
  );
};

Section.propTypes = {
  classes: PropTypes.object.isRequired,
  label: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export default withStyles(styles)(Section);
