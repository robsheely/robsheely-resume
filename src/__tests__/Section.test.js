import React from "react";
import { render, cleanup } from "react-testing-library";

import Section from "../Section";

const MOCK_DATA = {
  label: "Good Label",
  content: "Good Content"
};

describe("Section", () => {
  afterEach(cleanup);

  it("Passes smoke test", () => {
    const { container } = render(<Section name={name} label={MOCK_DATA.label}>
      <div>{MOCK_DATA.content}</div>
    </Section>);

    expect(container).toMatchSnapshot();
  });

  it("Renders label, divider and content when label is not null", () => {
    const { queryByTestId } = render(<Section label={MOCK_DATA.label}>
      <div>{MOCK_DATA.content}</div>
    </Section>);
    
const name = "good-label";

    expect(queryByTestId(`section-${name}`)).not.toBeNull();
    expect(queryByTestId(`section-${name}-content`)).not.toBeNull();
    expect(queryByTestId(`section-${name}-content-label`)).not.toBeNull();
    expect(queryByTestId(`section-${name}-content-label`).textContent).toBe(MOCK_DATA.label);
    expect(queryByTestId(`section-${name}-content-divider`)).not.toBeNull();
    expect(queryByTestId(`section-${name}-content`).children.length).toBe(3);
    expect(queryByTestId(`section-${name}-content`).children[ 2 ].textContent).toBe(MOCK_DATA.content);
  });

  it("Renders only content when label is null", () => {
    const { queryByTestId } = render(<Section>
      <div>{MOCK_DATA.content}</div>
    </Section>);

    const name = "none";
    
    expect(queryByTestId(`section-${name}`)).not.toBeNull();
    expect(queryByTestId(`section-${name}-content`)).not.toBeNull();
    expect(queryByTestId(`section-${name}-content-label`)).toBeNull();
    expect(queryByTestId(`section-${name}-content-divider`)).toBeNull();
    expect(queryByTestId(`section-${name}-content`).children.length).toBe(1);
    expect(queryByTestId(`section-${name}-content`).children[ 0 ].textContent).toBe(MOCK_DATA.content);
  });
});