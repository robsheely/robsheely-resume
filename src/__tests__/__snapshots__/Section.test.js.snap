// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Passes smoke test 1`] = `
<div>
  <div
    data-testid="section-good-label"
    id="section-good-label"
  >
    <div
      class="MuiPaper-root-7 MuiPaper-elevation1-10 MuiPaper-rounded-8 MuiCard-root-6 Section-card-1"
    >
      <div
        class="MuiCardContent-root-34 Section-content-2"
        data-testid="section-good-label-content"
        id="section-good-label-content"
      >
        <h6
          class="MuiTypography-root-35 Section-label-3 MuiTypography-h6-52 Section-h6-5 MuiTypography-colorInherit-64"
          data-testid="section-good-label-content-label"
          id="section-good-label-content-label"
        >
          Good Label
        </h6>
        <hr
          class="MuiDivider-root-71 Section-divider-4"
          data-testid="section-good-label-content-divider"
          id="section-good-label-content-divider"
        />
        <div>
          Good Content
        </div>
      </div>
    </div>
  </div>
</div>
`;
